pkg/booleanpolicy/violationmessages/printer/kube_event.go

```diff
@@ -82,16 +82,32 @@ func getDefaultViolationMsgHeader(event *storage.KubernetesEvent) string {
 	object := event.GetObject()
 	readableResourceName := strings.ToLower(object.Resource.String())
 
+	var singularResourceName string
+	if strings.HasSuffix(readableResourceName, "ies") {
+		singularResourceName = strings.TrimSuffix(readableResourceName, "ies")
+		singularResourceName = fmt.Sprintf("%sy", singularResourceName)
+	} else {
+		singularResourceName = strings.TrimSuffix(readableResourceName, "s")
+	}
+	singularResourceName = strings.ReplaceAll(singularResourceName, "_", " ")
+	readableResourceName = strings.ReplaceAll(readableResourceName, "_", " ")
+
+	var header string
 	if object.GetName() == "" {
-		return fmt.Sprintf("Access to %s in \"%s\"",
-			readableResourceName,
-			object.GetNamespace())
+		header = fmt.Sprintf("Access to %s", readableResourceName)
+		if object.GetNamespace() != "" {
+			header = fmt.Sprintf("%s in \"%s\"", header, object.GetNamespace())
```

Comment:
It is not clear to reader that it the namespace that is referenced. I recommend adding `namespace` to the message.